tests/functional/test_datarouter_plugin.py

```diff
@@ -176,25 +178,39 @@ def test_missing_creds_get_token_call(mock_method, datarouter_api_plugin):
                 "status_code": 200,
                 "content": 'This is A test',
             }
-        mock_method.return_value = json.dumps(op0)
+        mock_method.return_value = op0
         datarouter_api_plugin.provider_credentials = dict(dr_client_id='test',
                                                           host_url='https://test.url.com',)
-        with pytest.raises(KeyError,  match='dr_client_secret'):
-            datarouter_api_plugin.get_oauth_token(dr_token_url='https://test.url.com/token')
+        with pytest.raises(TypeError,  match='dr_client_secret'):
+            get_oauth_token(dr_token_url='https://test.url.com/token', dr_client_id='test')
 
     @staticmethod
     @mock.patch('teflo_datarouter_plugin.datarouter_plugin.requests.put')
     def test_send_put_req(mock_method, datarouter_api_plugin):
         """Test for PAYLOAD PUT req. GET TOKEN FOR TRACKING"""
 
         op0 = Response()
-        op0.status_code = 200
+        op0.status_code = 201
         op0._content = b'{"msg": "This is a test string 00000000-0000-0000-0000-000000000000"}'
 
         datarouter_api_plugin.provider_credentials = dict(host_url='https://test.url.com')
         mock_method.return_value = op0
         results = datarouter_api_plugin.send_put_req(access_token="testacceesstoke.com",
                                                      tar_payload={"path": "../assets/plugin_test.tar.gz"},
-                                                     json_config_file="../assets/user_config.json",
-                                                     dr_url='https://test.url.com')
+                                                     json_config_file="../assets/user_config.json")
         assert results == '00000000-0000-0000-0000-000000000000'
+
+    @staticmethod
+    @mock.patch('teflo_datarouter_plugin.helpers.requests.get')
+    def test_track_req_ok(mock_method, datarouter_api_plugin):
+        """Test for PAYLOAD PUT req. GET TOKEN FOR TRACKING"""
+
+        op0 = Response()
+        op0.status_code = 200
+        op0._content = b'{"status": "OK"}'
+
+        datarouter_api_plugin.provider_credentials = dict(host_url='https://test.url.com')
+        mock_method.return_value = op0
+        results = get_req_status(dr_client_id='test', dr_client_secret='secret', dr_token_url='https://api.test/toekn',
+                                 get_url='https://api.test/123-123-123', ac_token='token')
+        assert results['status'] == 'OK'
```

Comment:
Need to have a couple of tests when we see the error messages that the we waited for 5 min, and we got no response. so you can mock the count value to be 9, so the loop runs for one time and suggests that it ran 10 times and it failed to get any response.